### ISSUE TITLE
Freeze certain layers

### ISSUE THREAD

**Issue by mhasnat** [tflearn/tflearn#90]
Hi,

I was searching the TFLearn documents to freeze certain layers during CNN training. So far, I have found that only in the regression layer we can set the learning rate explicitly. Is there any parameter to fix the rate for other layers? Or is there any parameter to set the learning rate multiplier like caffe?

**Comment by aymericdamien** [comment]
Some optimizers accept a learning rate decay:
```
sgd = tflearn.SGD(learning_rate=0.1, lr_decay=0.96, decay_step=1000)
net = tflearn.regression(softmax, optimizer=sgd, metric='accuracy', loss='categorical_crossentropy')
```
Otherwise, as a hack, you may set multiple regression layers with different learning rate and you save/resume training every time you want to change learning rate.
About layers, unfortunately you can't now set a per layer custom learning rate.

**Comment by mhasnat** [comment]
Thank you very much for the answer.

Nice to know about the procedure to reduce learning rate automatically.

I am looking forward for the introduction of per-layer-wise learning rate set/control task.

**Comment by aymericdamien** [comment]
Yes, hope we can integrate that soon ;)
